dissect/target/plugins/os/windows/wer.py

```diff
@@ -11,6 +11,20 @@
 from dissect.target.target import Target
 
 camel_case_patterns = [re.compile(r"(\S)([A-Z][a-z]+)"), re.compile(r"([a-z0-9])([A-Z])"), re.compile(r"(\w)[.\s](\w)")]
+arraylike_pattern = re.compile(r"([_a-zA-Z][a-zA-Z0-9]*)\[(\d+)\]")
+
+
+def _clean_key(key: str) -> str:
+    def _clean(c):
+        return "_" + key[0].encode("unicode_escape").hex()
+    first = key[0] if (key[0].isidentifier() and key[0].isascii()) else _clean(key[0])
+
+    def _is_valid(c):
+        if c.isascii():
+            return c.isidentifier() or c.isdecimal()
+        return False
+
+    return first + ''.join([c if _is_valid(c) else _clean(c) for c in key[1:]])
```

Comment:
checking every character will most likely add some performance overhead, if there is no other way consider decorating the function with `@functools.cache`